modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/MuleApplicationPolicyProvider.java

```diff
@@ -68,8 +68,14 @@ public synchronized void addPolicy(PolicyTemplateDescriptor policyTemplateDescri
 
       Optional<RegisteredPolicyInstanceProvider> registeredPolicyInstanceProvider = registeredPolicyInstanceProviders.stream()
           .filter(p -> p.getPolicyId().equals(parametrization.getId())).findFirst();
+
       if (registeredPolicyInstanceProvider.isPresent()) {
-        throw new IllegalArgumentException(createPolicyAlreadyRegisteredError(parametrization.getId()));
+        if (isPolicyReorder(registeredPolicyInstanceProvider.get(), policyTemplateDescriptor, parametrization)) {
```

Comment:
add comments, please think on the poor soul that will run into this code in months to come.